processor/src/lib.rs

```diff
@@ -242,16 +245,28 @@ impl Process {
     fn execute_span_block(&mut self, block: &Span) -> Result<(), ExecutionError> {
         self.start_span_block(block)?;
 
+        let mut decorator_map_offset = 0;
+        let decorator_map = &block.get_decorator_map();
+        let mut decorator_iterator = DecoratorIterator::new(decorator_map);
```

Comment:
I'd probably just add `decorators()` method to the `Span` block which would return a new instance of `DecoratorIterator`.

Reply:
Does `decorator_iterator()` work as the name since we also need access to the decorator vector and the accessor method for that which can be named `decorators()`?

Reply:
Ah yes - let's use `decorator_iter()` then.

Reply:
Done